source/common/protobuf/message_validator_impl.cc

```diff
@@ -23,6 +23,10 @@ void onDeprecatedFieldCommon(absl::string_view description, bool soft_deprecatio
     throw DeprecatedProtoFieldException(absl::StrCat(description, deprecation_error));
   }
 }
+
+void onWorkInProgressCommon(absl::string_view description) {
+  ENVOY_LOG_MISC(warn, "{}", description);
```

Comment:
Is the idea to also bump a counter similar to what we do for deprecated features?

Reply:
Sure, that makes sense. I will add it.